tests/util/caches/test_descriptors.py

```diff
@@ -304,23 +315,23 @@ def check_result(r):
 
         return defer.gatherResults([d1, d2])
 
-    def test_cache_logcontexts_with_exception(self):
+    def test_cache_logcontexts_with_exception(self) -> Deferred:
```

Comment:
Are you using `Deferred` without an inner type to mean a Deferred that is never finished with a value>

Reply:
> It essentially is Deferred[Any] I think.

Yep, I think any omitted types for a generic type are assumed to be Any.

Reply:
I was able to make this one into `Deferred[None]` but a couple other spots run into annoying things with Deferreds needing to change types when they go through callbacks and that not really being supported with the type system.